src/main/java/com/minecolonies/coremod/permissions/ColonyPermissionEventHandler.java

```diff
@@ -206,6 +205,57 @@ public void on(final PlayerInteractEvent event)
     }
 
     /**
+     * PlayerInteractEvent.EntityInteract handler.
+     * <p>
+     * Check, if a player right clicked an entity.
+     * Deny if:
+     * - If the entity is in colony
+     * - player has not permission
+     *
+     * @param event PlayerInteractEvent
+     */
+    @SubscribeEvent
+    public void on(final PlayerInteractEvent.EntityInteract event)
+    {
+        final EntityPlayer playerIn = event.getEntityPlayer();
+        if (Configurations.enableColonyProtection && colony.isCoordInColony(playerIn.getEntityWorld(), playerIn.getPosition()))
+        {
+            final Permissions.Rank rank = colony.getPermissions().getRank(playerIn);
+
+            if (rank.ordinal() >= Permissions.Rank.FRIEND.ordinal())
```

Comment:
Still we would have to add them manually to the permission list. Which isn't loaded.
@Kostronor added a few new of these permissions on our server which never really worked because we kinda add them to the colony only once.
We should overwork the whole system probably.